### PR TITLE
Issue #6967 - symlog + minor ticks = exception

### PR DESCRIPTION
**Pull request by ColinDruery**
calling minorticks_on() with symlog previously caused an exception. This fix passes a transform object into the SymmetricalLogLocator instead of a float type. Appended tests to verify minorticks_on() works with various combinations of log and symlog. I believe this should solve the issue described in #6967.

### CODE REVIEW

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -5047,3 +5047,40 @@ def test_invalid_axis_limits():
         plt.ylim(np.nan)
     with pytest.raises(ValueError):
         plt.ylim(np.inf)
+
+
+# Test for x and y axes both symlog
+def test_minorticks_symlog():
+    plt.plot([1,2,3,4])
+    ax = plt.gca()
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_axes.py]: Better to use `plt.subplots`.

File: lib/matplotlib/tests/test_axes.py [hunk 2]
```diff
@@ -5047,3 +5047,40 @@ def test_invalid_axis_limits():
...
+
+
+# Test for x-axis symlog and y-axis log       
+def test_minorticks_different_logs2():
+    plt.plot([1,2,3,4])
+    ax = plt.gca()
+    ax.set_xscale('symlog')
+    ax.set_yscale('log')
+    ax.minorticks_on() 
+
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_axes.py]: Remove whitespace.

**Comment by phobson** [comment]
This looks good to me.

Totally optional, but if you're going to fix the first two commits, might I recommend also parametrizing your tests. So instead of the four tests you  currently have, I would do this:

```python
@pytest.mark.parametrize('xscale', ['symlog', 'log'])
@pytest.mark.parametrize('yscale', ['symlog', 'log'])
def test_minorticks_on(xscale, yscale):
    ax = plt.subplot(111)
    ax.plot([1, 2, 3, 4])  # note that I'm plotting with `ax`, not `plt`
    ax.set_xscale(xscale)
    ax.set_yscale(yscale)
    ax.minorticks_on()
```

**Comment by QuLogic** [comment]
Can you fix the email on the first two commits? There's a [script in GitHub's help](https://help.github.com/articles/changing-author-info/) for that. In step 5, make sure that you've only changed these 6 commits from this branch.

**Comment by ColinDruery** [comment]
Clearly I've made a mistake while fixing these first 2 commits. I'm going to open a new pull request with the correct information and code.

**Pull request closed without merging**